src/systems/variational_smoother_system.C

```diff
@@ -113,90 +114,124 @@ void VariationalSmootherSystem::prepare_for_smoothing()
-              target_elem.set_node(1, &node_1);
-              target_elem.set_node(2, &node_2);
+              for (const dof_id_type node_id : index_range(equilateral_points))
+                {
+                  // Scale the nodal positions to conserve area
+                  auto node_ptr = std::make_unique<Node>(
+                      equilateral_points[node_id] * side_length, node_id);
+                  target_elem->set_node(node_id, node_ptr.get());
+
+                  // Store the pointer so it stays alive
+                  owned_nodes.push_back(std::move(node_ptr));
+                }
 
               // build map
-              fe_map_target.init_reference_to_physical_map(dim, qrule_points,
-                                                           &target_elem);
-              fe_map_target.compute_map(dim, qrule_weights, &target_elem,
-                                        /*d2phi=*/false);
+              fe_map_target.init_reference_to_physical_map(dim, qrule_points, target_elem.get());
+              fe_map_target.compute_map(dim, qrule_weights, target_elem.get(), /*d2phi=*/false);
 
               // Yes, triangle-to-triangle mappings have constant Jacobians, but we
               // will keep things general for now
-              _target_inverse_jacobians[target_elem.type()] =
+              _target_inverse_jacobians[target_elem->type()] =
                   std::vector<RealTensor>(nq_points);
               for (const auto qp : make_range(nq_points))
-              {
-                const RealTensor H_inv =
-                    RealTensor(dxyzdxi[qp](0), dxyzdeta[qp](0), 0, dxyzdxi[qp](1),
-                               dxyzdeta[qp](1), 0, 0, 0, 1)
-                        .inverse();
-
-                _target_inverse_jacobians[target_elem.type()][qp] = H_inv;
-                target_elem_inverse_jacobian_dets[target_elem.type()][qp] =
-                    H_inv.det();
-              }
-
-              break;
-            }
-
-          default:
-            break;
-        }
-    }
-
-    Real elem_integrated_det_J(0.);
-    for (const auto qp : index_range(JxW))
-      elem_integrated_det_J +=
-          JxW[qp] * target_elem_inverse_jacobian_dets[elem->type()][qp];
-    const auto ref_elem_vol = elem->reference_elem()->volume();
-    elem_averaged_det_J_sum += elem_integrated_det_J / ref_elem_vol;
-
-  } // for elem
+                {
+                  const RealTensor H_inv =
+                      RealTensor(dxyzdxi[qp](0), dxyzdeta[qp](0), 0, dxyzdxi[qp](1),
+                                 dxyzdeta[qp](1), 0, 0, 0, 1)
+                          .inverse();
+
+                  _target_inverse_jacobians[target_elem->type()][qp] = H_inv;
+                  target_elem_inverse_jacobian_dets[target_elem->type()][qp] =
+                      H_inv.det();
+                }
+            }// if Tri
+        }// if find == end()
+
+      // Reference volume computation
+      Real elem_integrated_det_J(0.);
```

Comment:
Minor comment but I don't think we use this type of initialization for `Real`s elsewhere (we typically use boring `Real foo = 0.;`) so it looks a bit out of place IMO.